sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/structuredmessage/StructuredMessageEncoder.java

```diff
@@ -0,0 +1,258 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
One thing that I think we missed based on CI, it seems like we need to add a `package-info.java` to the new `structuredmessage` directory. We should be able to keep it minimal it looks by just declaring the package and having a docstring. I think we should be able to pattern match the other packages in the `implementation` directory.